app/constants/mods.py

```diff
@@ -53,113 +54,96 @@ def __repr__(self) -> str:
         if self.value == Mods.NOMOD:
             return "NM"
 
-        mod_str = []
-        _dict = mod2modstr_dict  # global
-
-        for mod in Mods:
-            if self.value & mod:
-                mod_str.append(_dict[mod])
-
+        mod_str = [mod2modstr_dict[mod] for mod in Mods if self.value & mod]
         return "".join(mod_str)
 
     def filter_invalid_combos(self, mode_vn: int) -> Mods:
         """Remove any invalid mod combinations."""
-
-        # 1. mode-inspecific mod conflictions
-        _dtnc = self & (Mods.DOUBLETIME | Mods.NIGHTCORE)
```

Comment:
note that i was previously only doing these types of things because of how expensive the IntFlag bitwise operations are as well - i was really trying to squeeze performance out wherever i could lol

Reply:
with bare ints, i wouldn't even bother as this hurts readability quite a bit (in bare ints, this would probably be ~100 nanoseconds)